### PR TITLE
Fix #1848

### LINKED ISSUE #1848: [NFR]Support `use` keyword in Closures
Example
```php
$func = function() use ($a) {
    return $a + 1;
}
```

Here `$a` becomes a local static variable of function `$func`.

@sergeyklay Who can implementation parse code.

### PR DESCRIPTION
**Pull request by dreamsxin**
Hello!

* Type: new feature
* Link to issue:

In raising this pull request, I confirm the following:

- [ ] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR
- [ ] I updated the CHANGELOG

Small description of change:

Thanks

### CODE REVIEW

**Comment by sergeyklay** [comment]
@dreamsxin Update .travis.yml to use `development` Zephir Parser

image

**Comment by dreamsxin** [comment]
@sergeyklay not supported PHP 5.x, and  AppVeyor build failed

**Comment by sergeyklay** [comment]
@dreamsxin Let's drop PHP 5.x support. I'll release Zephir 0.13.x for that.

**Comment by sergeyklay** [comment]
I'll fix AppVeyor a bit later

File: Library/Expression/Closure.php [hunk 1]
```diff
@@ -133,7 +152,39 @@ public function compile(array $expression, CompilationContext $compilationContex
...
+                    case 'uchar':
+                        $compilationContext->backend->assignLong($tempVariable, $var, $compilationContext);
+                        break;
+                    case 'double':
+                        $compilationContext->backend->assignDouble($tempVariable, $var, $compilationContext);
+                        break;
+                    case 'bool':
+                        $compilationContext->backend->assignBool($tempVariable, $var, $compilationContext);
+                        break;
+                    case 'string':
```
Comment (CameronHall) [inline Library/Expression/Closure.php:176]: This case is duplicated

Comment (dreamsxin) [reply Library/Expression/Closure.php:176]: Thank you,

**Comment by sergeyklay** [comment]
@dreamsxin I'll release php-zephir parser today. So that we'll able to fix AppVeyor. Could you remove php 5.6 from Travis build matrix ?

**Comment by sergeyklay** [comment]
@dreamsxin Could you please rebase. See: https://github.com/phalcon/zephir/pull/1852

**Comment by dreamsxin** [comment]
@sergeyklay Ok

**Comment by sergeyklay** [comment]
@dreamsxin Is it ready to merge?

**Comment by dreamsxin** [comment]
@sergeyklay You can see if there's anything else to optimize. For example, local static in symbal management.

File: Library/Expression/Closure.php [hunk 2]
```diff
@@ -133,7 +152,36 @@ public function compile(array $expression, CompilationContext $compilationContex
 
         $symbolVariable->initVariant($compilationContext);
         $compilationContext->backend->createClosure($symbolVariable, $classDefinition, $compilationContext);
-
+        $compilationContext->headersManager->add('kernel/object');
+        foreach ($staticVariables as $var) {
+            if ('variable' == $var->getType() || 'array' == $var->getType()) {
+                $compilationContext->backend->updateStaticProperty($classDefinition->getClassEntry(), $var->getName(), $var, $compilationContext);
+            } else {
+                $tempVariable = $compilationContext->symbolTable->getTempNonTrackedVariable('variable', $compilationContext, true);
```
Comment (sergeyklay) [inline Library/Expression/Closure.php:160]: @dreamsxin I think we'll use something like this:
```zep
fn () use (bar) {}
```
but not
```zep
fn () use (int bar) {}
```

By this, I think we don't need this branch. What do you think?

File: test/closures.zep [hunk 3]
```diff
@@ -48,4 +48,12 @@ class Closures
...
 		return x => (x + 100) + (x * 150);
 	}
+
+	public function testUseCommand()
+	{
+		var abc = 1;
+		return function() use (abc) {
+			return abc + 1;
+		};
+	}
```
Comment (sergeyklay) [inline test/closures.zep:58]: Could you create PHP test for that

**Review by sergeyklay — APPROVED** [review]
In general looks good. We're need more tests

File: Library/Expression/Closure.php (hunk 2 above)
Comment (dreamsxin) [reply Library/Expression/Closure.php:160]: zephir:
```zephir
var abc = 1;
return function() use (abc) {
	return abc + 1;
};
```
c:
```c
abc = 1;
zephir_create_closure_ex(return_value, NULL, xxx_x__closure_ce, SL("__invoke"));
ZEPHIR_INIT_ZVAL_NREF(tmp);
ZVAL_LONG(&tmp, abc);
zend_update_static_property(xxx_x__closure_ce, ZEND_STRL("abc"), &_0);
```

**Review by sergeyklay — COMMENTED** [review]
@dreamsxin Looks good to me

**Pull request merged**